### PR TITLE
rclone: build with 'cmount' support

### PR DESCRIPTION
**Pull request by midchildan**
###### Motivation for this change
`rclone cmount` provides the same functionality as `rclone mount`. They both provide FUSE integration, but `cmount` is backed by libfuse while `mount` is backed by a pure go reimplementation of libfuse.

`cmount` support is required for continued support for FUSE on macOS because the traditional `rclone mount` is [no longer supported for macOS][1]. Although `rclone mount` would work on macOS with recent versions of rclone, it's aliased to `rclone cmount`.

This change is not strictly required for Linux, but it still has its merits. According to the upstream maintainer, the libfuse implementation is "[a bit more compatible and reliable][2]" than the go implementation. Taking this into account, I've also enabled `cmount` support on Linux by default.

[1]: https://github.com/rclone/rclone/issues/4393
[2]: https://github.com/restic/restic/issues/3096#issuecomment-727215997

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [X] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [X] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by midchildan** [comment]
Just ran `<nixos/tests/nextcloud>` and `<nixos/tests/restic.nix>`, which relies on rclone. Both completed without problems.

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 117119` at 14d660be run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built successfully:</summary>
  <ul>
    <li><a href="https://gist.github.com/1666f4b98a0716ff5c5e8596e9e6b305">git-annex-remote-rclone</a></li></li>
    <li><a href="https://gist.github.com/a96655eda99c43c12ef854243281da4f">rclone</a></li></li>
    <li><a href="https://gist.github.com/95e94b8581eb817d4754fe08b9eec010">restic</a></li></li>
  </ul>
</details>

File: pkgs/applications/networking/sync/rclone/default.nix [hunk 1]
```diff
@@ -17,9 +20,14 @@ buildGoModule rec {
 
   outputs = [ "out" "man" ];
 
-  nativeBuildInputs = [ installShellFiles ];
+  buildInputs =
+    lib.optional enableCmount (if stdenv.isDarwin then macfuse-stubs else fuse);
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/sync/rclone/default.nix:24]:
```suggestion
  buildInputs = lib.optional enableCmount (if stdenv.isDarwin then macfuse-stubs else fuse);
```

Comment (midchildan) [reply pkgs/applications/networking/sync/rclone/default.nix]: Fixed.

File: pkgs/applications/networking/sync/rclone/default.nix [hunk 2]
```diff
@@ -17,9 +20,13 @@ buildGoModule rec {
 
   outputs = [ "out" "man" ];
 
-  nativeBuildInputs = [ installShellFiles ];
+  buildInputs = lib.optional enableCmount (if stdenv.isDarwin then macfuse-stubs else fuse);
+  nativeBuildInputs = [ installShellFiles makeWrapper ];
 
-  buildFlagsArray = [ "-ldflags=-s -w -X github.com/rclone/rclone/fs.Version=${version}" ];
+  buildFlagsArray = [
+    "-tags=cmount"
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/sync/rclone/default.nix:27]: This will not be disabled if I set enableCmount to false.

Comment (midchildan) [reply pkgs/applications/networking/sync/rclone/default.nix]: Missed that, thanks.

File: pkgs/applications/networking/sync/rclone/default.nix [hunk 3]
```diff
@@ -17,9 +20,11 @@ buildGoModule rec {
 
   outputs = [ "out" "man" ];
 
-  nativeBuildInputs = [ installShellFiles ];
+  buildInputs = lib.optional enableCmount (if stdenv.isDarwin then macfuse-stubs else fuse);
+  nativeBuildInputs = [ installShellFiles makeWrapper ];
 
-  buildFlagsArray = [ "-ldflags=-s -w -X github.com/rclone/rclone/fs.Version=${version}" ];
+  buildFlagsArray = lib.optional enableCmount "-tags=cmount"
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/sync/rclone/default.nix:26]:
```suggestion
  buildFlagsArray = lib.optionals enableCmount [ "-tags=cmount" ]
```

**Pull request merged**